### ISSUE TITLE
[research: 1d] Figure out if app.internal domains need explicit TLS configuration

### ISSUE THREAD

**Issue by mogul** [GSA/data.gov#2945]
## User Story

In order to ensure we know how to meet requirements for TLS on traffic between applications, the data.gov team wants to know how/if app.internal traffic is secured by the cloud.gov platform, and if there are additional requirements for us.

## Acceptance Criteria
[ACs should be clearly demoable/verifiable whenever possible. Try specifying them using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications).]
- [x] WHEN I look at this issue \
  THEN I see a summary of what's actually required for our ATO \
  AND I see a link to an issue summarizing the technical effort needed to secure the traffic if necessary.

## Background
[Any helpful contextual notes or links to artifacts/evidence, if needed]

Although traffic must explicitly be allowed between containers in order to make use of app.internal routes, Bret's understanding is that the traffic may travel in the clear between Diego cells, and there's not a platform-provided way to add TLS there. We'd love to be proven wrong! This story is about clarifying whether additional measures are needed to prevent traffic from being in the clear when traveling between containers internal to cloud.gov, and characterizing the relative level of risk for our ISSM.

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))
[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
[Any security concerns that might be implicated in the change. "None" is OK, just be explicit here!]

## Sketch
[Notes or a checklist reflecting our understanding of the selected approach]

One option is to just put public routes and auth in place to avoid having to hassle with TLS termination/rotation ourselves.

Another would be to run the backend app as a sidecar/secondary app for a TLS termination proxy (eg `nginx_buildpack`).

**Comment by mogul** [comment]
cloud.gov is hosting [a tech talk on Wednesday of next week](https://www.eventbrite.com/e/tech-talk-internal-routes-tickets-144475338627) about this topic... We should attend and see if we can get the question answered there.

**Comment by mogul** [comment]
The tech talk was good, but didn't address our concerns. We've asked for [followup with the cloud.gov team in Slack](https://gsa-tts.slack.com/archives/C09CR1Q9Z/p1616013890003900).

**Comment by mogul** [comment]
No loner waiting on advice from cloud.gov. Will write another issue capturing our potential solution.

**Comment by mogul** [comment]
We should have some guidance as to how necessary this work is likely to be after Friday's ISSM meeting.

**Comment by mogul** [comment]
Ryan says that yes, we should be trying to encrypt inter-app traffic as much as possible, even if it's in the cloud.gov VPC and on an overlay network with allow-list semantics. The reason is to frustrate lateral compromise attempts, security-in-depth style. (He referenced SolarWinds.)

I've identified one approach I'm comfortable with for doing this: Deploy an `nginx-buildpack` proxy app which uses OpenResty to [auto-rotate the TLS certificate](https://github.com/auto-ssl/lua-resty-auto-ssl) on the app.internal domain. Then deploy the **_actual_** app as a sidecar to the proxy app. [Here's the issue about that.](https://github.com/GSA/datagov-deploy/issues/3035)

There's an alternative opportunity to [replace datapusher with xloader](https://github.com/GSA/datagov-deploy/issues/1486) ([PR already complete](https://github.com/GSA/inventory-app/pull/29)), in which case we will no longer have anything on the app.internal domain, so figuring out how to secure it is therefore no longer necessary! We've moved that issue into `New` so we'll discuss that trade-off in the next grooming meeting.
